framework/include/materials/MaterialPropertyStorage.h

```diff
@@ -255,13 +255,13 @@ class MaterialPropertyStorage
     return _prop_names.count(retrievePropertyId(prop_name)) > 0;
   }
 
+  /// Remove the property storage and element pointer from internal data structures
+  void eraseProperty(const Elem * elem);
```

Comment:
Maybe add a sentence to the doc comment saying "Use this when elements are deleted so we don't end up with invalid elem pointers (for e.g. stateful properties) hanging around in our data structures." - or something like that

Reply:
I will add what you suggest. Thanks.